src/ansys/mapdl/core/xpl.py

```diff
@@ -305,9 +345,10 @@ def read(self, recordname):
         array([ 4,  7, 10, 13, 16, 19, 22, 25, 28, 31, 34, 37, 40, 43,
                46, 49, 52, 55, 58,  1], dtype=int32)
         """
-        response = self._mapdl.run("*XPL,READ,%s,TmpXplData" % recordname)
+        rand_name = id_generator()
+        response = self._mapdl.run("*XPL,READ,%s,rand_name" % recordname)
         self._check_ignored(response)
-        response = self._mapdl._data_info("TmpXplData")
```

Comment:
Change of the hard-coded name by a random Id to avoid name conflicts when calling several times